RULES.md

```diff
@@ -584,6 +585,24 @@ A row in the input file has only spaces.
 ##### References:
 * [GTFS file requirements](http://gtfs.org/reference/static/#file-requirements)
 
+<a name="FastTravelBetweenConsecutiveStopsNotice"/>
+
+#### FastTravelBetweenConsecutiveStopsNotice
+
+A transit vehicle moves too fast between two consecutive stops. The speed threshold depends on route type.
+
+##### References:
+* [Original Python validator implementation](https://github.com/google/transitfeed)
+
+#### FastTravelBetweenFarStopsNotice
+
+A transit vehicle moves too fast between far consecutive stops (more than in 10 km apart). 
+This normally indicates a more serious problem than too fast travel between consecutive stops.
```

Comment:
If this is a "more serious problem", should it be elevated to an error instead of warning?

Reply:
I do not want to reject 225 feeds that have this problem, so I would like to keep it as warning.

By the way, can you run this validator on several thousands feeds that you have in MobilityData?

Reply:
Makes sense. 

@lionel-nj would know more about the state of running against the large feed database. Related, I noticed that even the end-to-end-100 job is failing to complete on recent commits - https://github.com/MobilityData/gtfs-validator/runs/3246259762.

Reply:
I see, the Validator was out of memory. May we give it 3 GiB? I think it is usually enough.

Reply:
Validator was out of memory because se-trafiklab reports trains using extended vehicle type (100, 101, 102 etc). The upstream validator does not support extended vehicle types - it shows a warning for unknown enum value. StopTimeTravelSpeedValidator uses 50 km/h threshold for "unknown" vehicle types which is too low for trains, so GTFS Validator produces too many warnings for "fast travel" and gets out of memory.

Mitigation. Use 200 km/h threshold for unknown vehicle types: https://github.com/MobilityData/gtfs-validator/pull/946

Solution. Add support for extended route types. At Google. we convert them to basic route types.

```java
/**
   * Converts GTFS route type to a narrow range of basic types.
   *
   * <p>Google supports extended route types (Hierarchical Vehicle Type, HVT) that are not a part of
   * GTFS standard but they are widely used. This function converts HVT to basic route types.
   *
   * <p>Google also supports non-standard route types (intercity bus and commuter train) that are
   * also converted to basic ones.
   */
  public static BasicType toBasicType(GtfsRouteType routeType) {
    switch (routeType) {
      case LIGHT_RAIL:
        return BasicType.LIGHT_RAIL;
      case SUBWAY:
        return BasicType.SUBWAY;
      case RAIL:
      case COMMUTER_TRAIN:
        return BasicType.RAIL;
      case INTERCITY_BUS:
      case BUS:
      case HVT_COMMUNAL_TAXI:
        return BasicType.BUS;
      case FERRY:
        return BasicType.FERRY;
      case CABLE_TRAM:
        return BasicType.CABLE_TRAM;
      case AERIAL_LIFT:
        return BasicType.AERIAL_LIFT;
      case FUNICULAR:
        return BasicType.FUNICULAR;
      case TROLLEYBUS:
        return BasicType.TROLLEYBUS;
      case MONORAIL:
      case HVT_MONORAIL:
        return BasicType.MONORAIL;
      case HORSE_CARRIAGE:
      case HVT_HORSE_DRAWN_CARRIAGE:
        return BasicType.HORSE_CARRIAGE;
      default:
        break;
    }

    // Support for Hierarchical Vehicle Types.
    switch (routeType.getNumber() / 100) {
      case 1:
        return BasicType.RAIL;
      case 2:
      case 7:
        return BasicType.BUS;
      case 4:
        return BasicType.SUBWAY;
      case 8:
        return BasicType.TROLLEYBUS;
      case 9:
        return BasicType.LIGHT_RAIL;
      case 10:
      case 12:
        return BasicType.FERRY;
      case 13:
        return BasicType.AERIAL_LIFT;
      case 14:
        return BasicType.FUNICULAR;
      default:
        return BasicType.MISCELLANEOUS;
    }
  }
```

Reply:
> By the way, can you run this validator on several thousands feeds that you have in MobilityData?

The process is under construction: https://github.com/MobilityData/gtfs-validator/pull/848 & https://github.com/MobilityData/gtfs-validator/pull/933. We are considering using Kubernetes to run the validator on all datasets available on the mobility archives (1300+). 

> I see, the Validator was out of memory. May we give it 3 GiB? I think it is usually enough.

Yes, see #947.